### PR TITLE
Add automatic slideshow

### PR DESCRIPTION
**Pull request by DominicWindisch**
__This PR adds an automatic slideshow.__

The slideshow starts after the user is idle for a configurable timeout (UI Settings: TIMEOUT_TO_SLIDESHOW). Can be set to 0 to disable automatic slideshow.

To monitor the idle time, a `RouteAfterTimeout` component is added. This will automatically route to a path after a configured timeout. 10 seconds before the timeout, a warning is displayed to the user. Interacting with the page will cancel the warning and reset the idle timer.

Currently, the slideshow starts automatically from
- index page
- gallery page
- item presenter page
- item approval page (from my testing, auto-routing back to index in the middle of the process does not brake the state machine)

The slideshow itself is implemented in the `SlideshowPage` which just displays a galleryitemdetail component in fullscreen (i.e. without the toolbar and QR code). The back button will then bring users back to the starting page. I opted for a back button such that users can interact with the slideshow (swipe etc) if they want to instead of automatically taking them back to the main page on any interaction.

As of now, the slideshow
- uses "fade" animation between items
- autoplays with 5 seconds per image
- displays items in random order

### CODE REVIEW

**Comment by mgineer85** [comment]
The corresponding part in -app with TIMEOUT_TO_SLIDESHOW is missing in my dev.

Since the UI and the frontend are quite tight connected to each other, I consider to move frontend to app to have everything in one place.   What do you think about this idea?

File: package.json
Comment (mgineer85) [inline package.json:1]: vueuse seems like a reasonable package. 
not sure yet which duplicates come with it using
-native fetch api and
-quasar framework
-vuesse
We can keep it, maybe one day we can unify to one dependency.

**Comment by mgineer85** [comment]
I like this addition very much, thank you :)


There seems to be some leftovers from toolbar on top of the image.
Also the image is not scaled correctly yet to 100% window height.
image

**Comment by mgineer85** [comment]
Not sure yet if it's just for me.
-The background is pinkish - I would expect just a darkened background as with other dialog boxes.
-if the timeout_to_slideshow is smaller than the abort time, you get stuck with the message display

The message dialog is quite big text-wise. I would expect somebody to stand right before the box if he wants to abort.
Could also consider a notifier like this: https://quasar.dev/quasar-plugins/notify#timeout-progress

image

If you find time to improve, would be great, otherwise I will do later when I have time :)

**Comment by DominicWindisch** [comment]
> Since the UI and the frontend are quite tight connected to each other, I consider to move frontend to app to have everything in one place. What do you think about this idea?

Sure, certainly easier than manually synching every change. 👍

I'm currently targeting an iPad Mini with the layout, so there's likely some quirks when viewing on other aspect ratios etc. 😬 But there are quite a few quirks in the frontend which make it look odd on some form factors/resolutions so I didn't mind fine tuning too much as of now. E.g., for me, the fixed pixel sizes of many elements don't fit my resolution at all, so I'm overriding them with vh/vw. 

I've missed adding an ID to the slideshow elements; that would also help with styling for now.

The pinkish background is btw based on the secondary color (50% opacity). Again, an ID would help with styling overrides...

**Comment by DominicWindisch** [comment]
The timeout notifier looks great 👍 Gonna change to that when I find the time. Likely no earlier than Friday, though.

**Comment by mgineer85** [comment]
> I'm currently targeting an iPad Mini with the layout, so there's likely some quirks when viewing on other aspect ratios etc. 😬 But there are quite a few quirks in the frontend which make it look odd on some form factors/resolutions so I didn't mind fine tuning too much as of now. E.g., for me, the fixed pixel sizes of many elements don't fit my resolution at all, so I'm overriding them with vh/vw.

maybe you can create another issue with screenshots?
I use one large monitor during dev but also two smaller touchscreens with 7 and 10 inch which is looking fine.

**Comment by DominicWindisch** [comment]
> maybe you can create another issue with screenshots?

Sure, I'll create one once I gathered some screenshots. On that note, I guess you only target landscape orientation? (Which is fine for me just so I don't bother you with portrait orientation oddities)

**Comment by DominicWindisch** [comment]
Should be good now. Here's the backend change: https://github.com/photobooth-app/photobooth-app/pull/187

**Comment by mgineer85** [comment]
Hey, just want to test in combination with #187 but seems it's not working?
Does it work for you on latest pullrequest?

**Comment by DominicWindisch** [comment]
Yes, that worked. Didn't update web spa on the app commit though.

**Pull request merged**